core/invoker/src/main/scala/whisk/core/containerpool/ContainerPool.scala

```diff
@@ -115,26 +115,31 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
 
     // Container is free to take more work
     case NeedWork(data: WarmedData) =>
-      freePool.update(sender(), data)
-      busyPool.remove(sender()).foreach(_ => feed ! MessageFeed.Processed)
+      freePool = freePool + (sender() -> data)
+      busyPool = busyPool - sender()
+      busyPool.foreach { _ =>
```

Comment:
The logic here is wrong. Above code said: "remove this element, and if it was actually there, also send a message". So it's going to be more like:

```scala
if(busyPool.contains(sender()) {
  busyPool = busyPool - sender()
  feed ! MessageFeed.Processed
}
```

Reply:
oof. that was a stupid bug, but explains the behavior I was seeing in my failed tests.